### PR TITLE
add flags to velocity and position estimate abi messages

### PR DESCRIPTION
**Pull request by kirkscheper**
This makes it optional to define if you want to update the horizontal and/or vertical parameters of the velocity and position estimates from the abi message. This should address the issue #2144 .

### CODE REVIEW

**Comment by gautierhattenberger** [comment]
In `guidance_opticflow_hover.c` line 100 the declaration of the callback is not updated

**Comment by kirkscheper** [comment]
Wait till #2132 is accepted before merging this. Then I can extend the fixes to that module as well.

**Comment by tomvand** [comment]
Thanks Kirk, really needed this :).

Would it be prudent to add separate bits for the X and Y axes? (In case someone builds a forward-only velocity estimator?)

**Comment by kirkscheper** [comment]
Not really. I can't image a situation where you would only estimate forward velocity with a rotorcraft, you would need both to be able to fly stably. In the odd case that that happens this can always be expanded later.

**Comment by tomvand** [comment]
I agree for rotorcraft, but aren't the ABI messages also used for fixed-wing/hybrid uavs? Expanding later is fine by me btw.

**Comment by flixr** [comment]
If we already add flags, I would really propose to add them for each axis.
Another maybe even better option would be to extend the noise, i.e. create a `noise_x`, `noise_y`, `noise_z`. That way you could even have a different measurement/estimate noise for each axis... and to mark one axis as invalid, just set it to a negative value.

**Comment by kirkscheper** [comment]
I went with @flixr comment. I think this is done now.

**Comment by tomvand** [comment]
Looks good

File: sw/airborne/modules/guidance_opticflow/guidance_opticflow_hover.c [hunk 1]
```diff
@@ -149,28 +149,42 @@ void guidance_h_module_run(bool in_flight)
...
+  if (noise_x >= 0.f)
+  {
+    /* Calculate the error */
+    float err_vx = opticflow_stab.desired_vx - vel_x;
 
-  /* Calculate the integrated errors (TODO: bound??) */
-  opticflow_stab.err_vx_int += err_vx / 512;
-  opticflow_stab.err_vy_int += err_vy / 512;
+    /* Calculate the integrated errors (TODO: bound??) */
+    opticflow_stab.err_vx_int += err_vx / 512;
```
Comment (gautierhattenberger) [inline sw/airborne/modules/guidance_opticflow/guidance_opticflow_hover.c]: what is this 512 ? If it is the related to the general periodic freq, it shouldn't be hard-coded. Or at least configurable.

Comment (kirkscheper) [reply sw/airborne/modules/guidance_opticflow/guidance_opticflow_hover.c]: I guess it probably is but I didn't change this, only added a tab.

IMHO this entire module doesn't make much sense. This functionality should be possible with just normal control loops. I also see that this module isn't included in any airframe. I am in favour of deleting this altogether.

Comment (gautierhattenberger) [reply sw/airborne/modules/guidance_opticflow/guidance_opticflow_hover.c]: I don't mind deleting old code. I guess it was a demo module made by Christophe or Freek. I let you guys decide if it is worth keeping it.

**Pull request merged**